### ISSUE TITLE
App Configuration resources

### ISSUE THREAD

**Issue by Sahasrara** [terraform-providers/terraform-provider-azurerm#3514]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please create app_configuration resources so that app configuration stores can be Terraformed.

### New or Affected Resource(s)

* azurerm_app_configuration_store

### Potential Terraform Configuration

```
resource "azurerm_app_configuration_store" "app_config" {
  name                = 
  subscription   = 
  location            =
  resource_group_name =
}
```

**Comment by brennerm** [comment]
Started implementing the new resource at #4859. Feel free to add your comments/wishes there.

**Comment by mbfrahry** [comment]
Thanks for opening this @Sahasrara and for doing the legwork @brennerm.

This has been released in [version 1.39.0 of the provider](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.39.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading. As an example:
```hcl
provider "azurerm" {
    version = "~> 1.39.0"
}
# ... other configuration ...
```
